lib/exposer.cc

```diff
@@ -4,16 +4,23 @@
 
 #include "prometheus/exposer.h"
 
+#include "CivetServer.h"
+#include "handler.h"
 #include "metrics.pb.h"
 
 namespace prometheus {
 
 Exposer::Exposer(const std::string& bind_address)
-    : server_({"listening_ports", bind_address.c_str()}),
+    : server_(new CivetServer{{"listening_ports", bind_address.c_str()}}),
       exposer_registry_(std::make_shared<Registry>()),
-      metrics_handler_(collectables_, *exposer_registry_) {
+      metrics_handler_(
+          new detail::MetricsHandler{collectables_, *exposer_registry_}) {
   RegisterCollectable(exposer_registry_);
-  server_.addHandler("/metrics", &metrics_handler_);
+  server_->addHandler("/metrics", metrics_handler_.get());
+}
+
+Exposer::~Exposer() {
+  // needed to properly destroy server_
```

Comment:
We should either call removeHandler() or reverse the order of the `server_` and metrics_handler_` members. The way it is right now, there's a race condition where `metrics_handler_` is destroyed and `server_` dispatches to its dangling pointer.